profiling/src/profile/mod.rs

```diff
@@ -942,6 +987,52 @@ impl Profile {
 
         Ok(new_values)
     }
+
+    /// Given a sample, and a set of observations at different timestamps,
+    /// expand to one Sample for each timestamp.
+    fn expand_timestamped_sample(
+        &self,
+        sample: &Sample,
+        observations: &[(Timestamp, Vec<i64>)],
+    ) -> anyhow::Result<Vec<pprof::Sample>> {
```

Comment:
I wonder if we could reuse more of the code for this step when comparing to the timestampless variant?

Reply:
I made a helper function to reuse code